File: dotnet/cafdemalihapa/Names/DirectoryToBeOpen.cs
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Names
{
    public class DirectoryToBeOpen(
        ILogger<DirectoryToBeOpen> logger
        )
    {
        public string GetPath()
        {
            var directoryToBeOpen = CommandLineArgs.GetByKey("--directory-to-be-open");
            logger.LogInformation("directory to be open: {directoryToBeOpen}", directoryToBeOpen);
            return directoryToBeOpen;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Names/DirectoryToBeOpen.fs
namespace cafdemwimu.console.Names

open Microsoft.Extensions.Logging
open cafdemwimu.console

type DirectoryToBeOpen(logger: ILogger<DirectoryToBeOpen>) =
    member _.GetPath() =
        let directoryToBeOpen = CommandLineArgs.GetByKey("--directory-to-be-open")
        logger.LogInformation("directory to be open: {directoryToBeOpen}", directoryToBeOpen)
        directoryToBeOpen
